### PR TITLE
[pytest] Convert FDB MAC Expire Test

### PR DESCRIPTION
**Pull request by tahmed-dev**
### Description of PR
Summary:
Convert FDB MAC expire test from Ansible to pytest.

signed-off-by: Tamer Ahmed <tamer.ahmed@microsoft.com>

Fixes # (issue)
### Type of change
- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Approach
#### What is the motivation for this PR?

#### How did you do it?

#### How did you verify/test it?
```
tests$ pytest fdb/test_fdb_mac_expire.py --testbed=vms12-t0-s6000-1 --inventory=../ansible/str --testbed_file=../ansible/testbed.csv --host-pattern=str-s6000-acs-14 --module-path=../ansible/library --disable_loganalyzer
================================================================================================================================================== test session starts ===================================================================================================================================================
platform linux2 -- Python 2.7.12, pytest-4.6.9, py-1.8.1, pluggy-0.13.1
ansible: 2.8.7
rootdir: /var/host-acs-mgmt-repo/tests, inifile: pytest.ini
plugins: ansible-2.2.2
collected 1 item                                                                                                                                                                                                                                                                                                         

fdb/test_fdb_mac_expire.py .                                                                                                                                                                                                                                                                                       [100%]

========================================================================================================================================= 1 passed, 1 warnings in 218.94 seconds =========================================================================================================================================
```
and
```
tests$ pytest fdb/test_fdb_mac_expire.py --testbed=vms2-2-t0-2700 --inventory=../ansible/str --testbed_file=../ansible/testbed.csv --host-pattern=str-msn2700-20 --module-path=../ansible/library --disable_loganalyzer
================================================================================================================================================== test session starts ==================================================================================================================================================
platform linux2 -- Python 2.7.12, pytest-4.6.9, py-1.8.1, pluggy-0.13.1
ansible: 2.8.7
rootdir: /var/host-acs-mgmt-repo/tests, inifile: pytest.ini
plugins: ansible-2.2.2
collected 1 item                                                                                                                                                                                                                                                                                                        

fdb/test_fdb_mac_expire.py .                                                                                                                                                                                                                                                                                      [100%]

======================================================================================================================================== 1 passed, 1 warnings in 206.10 seconds =========================================================================================================================================
```
#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation 
<!--
(If it's a new feature, new test case)
Did you update documentation/Wiki relevant to your implementation?
Link to the wiki page?
-->

### CODE REVIEW

File: tests/fdb/test_fdb_mac_expire.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+                request (Fixture): pytest request object
+                testbed (Fixture, dict): Map containing testbed information 
+                duthost (AnsibleHost): Device Under Test (DUT)
+                ptfhost (AnsibleHost): Packet Test Framework (PTF)
+
+            Returns:
+                None
+        """
+        topo = testbed["topo"]["name"]
+        if topo not in self.SUPPORTED_DUT_TOPOS:
```
Comment (yxieca) [inline tests/fdb/test_fdb_mac_expire.py:228]: You could check testbed['topo']['type'] to be 't0'. Should all T0 topology support this test?

File: tests/fdb/test_fdb_mac_expire.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+        count = 0
+        dummyMacCount = self.__getFdbTableCount(duthost, self.DUMMY_MAC_PREFIX)
+        while count * self.POLLING_INTERVAL_SEC <= fdbAgingTime + self.FDB_AGING_TIMER_EXPIRE_WINDOW and \
+            dummyMacCount != 0:
+            time.sleep(self.POLLING_INTERVAL_SEC)
+            dummyMacCount = self.__getFdbTableCount(duthost, self.DUMMY_MAC_PREFIX)
+            count += 1
+
+        assert dummyMacCount == 0 and count * self.POLLING_INTERVAL_SEC >= fdbAgingTime, \
```
Comment (yxieca) [inline tests/fdb/test_fdb_mac_expire.py:250]: Please use pytest_assert instead

File: tests/fdb/args/fdb_args.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        Returns:
+            None
+    """
+    fdb_group = parser.getgroup("FDB test suite options")
+
+    fdb_group.addoption(
+        "--fdb_aging_time",
+        action="store",
+        type=int,
+        default=60,
```
Comment (yxieca) [inline tests/fdb/args/fdb_args.py:19]: I think the default should be 600 seconds

File: tests/fdb/test_fdb_mac_expire.py (hunk 1 above)
Comment (tahmed-dev) [reply tests/fdb/test_fdb_mac_expire.py:228]: Will change it. Re: supported topo'ss, these are the supported T0 topo's:
```
    SUPPORTED_DUT_TOPOS = ["t0", "t0-64", "t0-64-32", "t0-116", "t0-52"]
```
These is subset of the T0 topo's. Will check for T0 only as there nothing should prevent this from running on all T0 variants.

File: tests/fdb/test_fdb_mac_expire.py (hunk 2 above)
Comment (tahmed-dev) [reply tests/fdb/test_fdb_mac_expire.py:250]: I will change it to pytest_assert.

File: tests/fdb/args/fdb_args.py (hunk 3 above)
Comment (tahmed-dev) [reply tests/fdb/args/fdb_args.py:19]: Ansible test does change the default timeout to be 60 seconds [here](https://github.com/Azure/sonic-mgmt/blob/master/ansible/roles/test/tasks/fdb_mac_expire.yml#L9) for testing purpose. The default value is restored from the backup copy of switch.json.

**Comment by daall** [comment]
retest vsimage please

**Comment by daall** [comment]
retest vsimage please

File: tests/fdb/test_fdb_mac_expire.py [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+
+            Args:
+                duthost (AnsibleHost): Device Under Test (DUT)
+                mac (str): MAC value to search for in the FDB table
+
+            Returns:
+                (int) representing the number of the FDB entries containing th eMAC address
+        """
+        result = duthost.shell("show mac | grep {0} | wc -l".format(mac))
+        pytest_assert(result["rc"] == 0, "Failed to obtain mac address with error '{0}'".format(result["stderr"]))
```
Comment (wangxin) [inline tests/fdb/test_fdb_mac_expire.py:34]: If run the above shell module without 'module_ignore_errors=True' argument, then exception will be raised if rc!=0. The pytest_assert would not get a chance to be executed here when rc!=0.

File: tests/fdb/test_fdb_mac_expire.py [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+        """
+        result = duthost.shell(argv=[
+            "docker",
+            "exec",
+            "swss",
+            "bash",
+            "-c",
+            "swssconfig /etc/swss/config.d/switch.json"
+        ])
+        pytest_assert(result["rc"] == 0, "Failed to load swss config with error '{0}'".format(result["stderr"]))
```
Comment (wangxin) [inline tests/fdb/test_fdb_mac_expire.py:59]: Similar issue.

File: tests/fdb/test_fdb_mac_expire.py [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+            "--relax",
+            "--debug",
+            "info",
+            "--log-file",
+            "/tmp/{0}".format(testCase)
+            ],
+            chdir = "/root",
+            )["rc"] == 0,
+            "Failed when running test '{0}'".format(testCase)
+        )
```
Comment (wangxin) [inline tests/fdb/test_fdb_mac_expire.py:110]: Similar issue. If rc of ptfhost.shell is not 0, a exception will be raised. There is no chance to check `rc` value here.

**Pull request merged**